### PR TITLE
snapshot deployment

### PR DESCRIPTION
**Pull request by digitalbuddha**
global:   - secure in `travis.yml` needs to be updated to your keys @ansman 

LMK if there's any copy pasta I missed :-)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,22 @@
 language: java
 jdk:
-  - oraclejdk8
\ No newline at end of file
+- oraclejdk8
+
+script:
+- ./gradlew check --stacktrace
```
Comment (JakeWharton) [inline .travis.yml]: Use `build`. Otherwise you skip packaging and cause it to happen during the deploy step.

File: .travis.yml [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+  - gradle/deploy_snapshot.sh
+branches:
+  except:
+  - gh-pages
+notifications:
+  email: false
+sudo: false
+cache:
+  directories:
+  - $HOME/.m2
```
Comment (JakeWharton) [inline .travis.yml]: Unused

File: gradle/deploy_snapshot.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+#
+# Deploy a jar, source jar, and javadoc jar to Sonatype's snapshot repo.
+#
+# Adapted from https://coderwall.com/p/9b_lfq and
+# http://benlimmer.com/2013/12/26/automatically-publish-javadoc-to-gh-pages-with-travis-ci/
+
+SLUG="Ansman/Kotshi"
```
Comment (JakeWharton) [inline gradle/deploy_snapshot.sh]: Lowercase? Isn't Travis CI case sensitive on these?

File: gradle/deploy_snapshot.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+JDK="oraclejdk8"
+BRANCH="feature/rx2"
```
Comment (JakeWharton) [inline gradle/deploy_snapshot.sh]: `master`

File: gradle/bintray.gradle [hunk 5]
```diff
@@ -1,5 +1,5 @@
 apply plugin: 'com.jfrog.bintray'
-apply plugin: 'com.github.dcendents.android-maven'
+//apply plugin: 'com.github.dcendents.android-maven'
```
Comment (digitalbuddha) [inline gradle/bintray.gradle]: this plugin clashes with the `maven-push.gradle` from what I gather it is due to both trying to bring in an install task.  I can't seem to find why `com.github.dcendents.android-maven` is needed unless bintray deployment depends on it.

File: gradle/bintray.gradle [hunk 6]
```diff
@@ -38,64 +38,3 @@ bintray {
         }
     }
 }
-
-afterEvaluate { project ->
```
Comment (ansman) [inline gradle/bintray.gradle]: I don't think that deploying to bintray will work without this?

File: gradle.properties [hunk 7]
```diff
@@ -2,7 +2,7 @@ GROUP=se.ansman.kotshi
 VERSION_NAME=0.3.0-SNAPSHOT
```
Comment (ansman) [inline gradle.properties]: Is it possible to overwrite this in the upload script? I would like to append the current time.

Comment (JakeWharton) [reply gradle.properties]: That's usually done automatically on the receiving side

Comment (ansman) [reply gradle.properties]: Oh, even better.

File: gradle/bintray.gradle (hunk 6 above)
Comment (digitalbuddha) [reply gradle/bintray.gradle]: Same block is coming in from the maven deploy script.

Comment (ansman) [reply gradle/bintray.gradle]: Awesome

**Comment by digitalbuddha** [comment]
Don't forget to change the keys

**Pull request merged**